pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -104,9 +124,24 @@ def handle_text_delta(
             part_index = self._vendor_id_to_part_index.get(vendor_part_id)
             if part_index is not None:
                 existing_part = self._parts[part_index]
-                if not isinstance(existing_part, TextPart):
+
+                if extract_think_tags and isinstance(existing_part, ThinkingPart):
+                    # We may be building a thinking part instead of a text part if we had previously seen a `<think>` tag
+                    if content == END_THINK_TAG:
```

Comment:
Why is it a ==? Can't the content just have the end tag? I didn't follow.

Reply:
@Kludex Yep as far as I can tell!